File: php/src/Model/WorkerLoad.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * One worker backlog against their cap.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkerLoad
{
    public function __construct(
        public readonly string $workerId,
        public readonly int $backlog,
        public readonly int $maxBacklogSize,
        public readonly bool $available,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            workerId: (string) ($data['workerId'] ?? ''),
            backlog: (int) ($data['backlog'] ?? 0),
            maxBacklogSize: (int) ($data['maxBacklogSize'] ?? 0),
            available: (bool) ($data['available'] ?? false),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/WorkerLoad.java
package io.fivexer.sdk.model;

/**
 * One worker's backlog against their cap.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkerLoad {
    private String workerId;
    private int backlog;
    private int maxBacklogSize;
    private boolean available;
    // Same distinction as on WorkerDetail: an invite nobody accepted, or a join awaiting approval
    private boolean invitePending;
    private boolean pendingApproval;

    public String getWorkerId() { return workerId; }
    public int getBacklog() { return backlog; }
    public int getMaxBacklogSize() { return maxBacklogSize; }
    public boolean isAvailable() { return available; }
    public boolean isInvitePending() { return invitePending; }
    public boolean isPendingApproval() { return pendingApproval; }
}
